### PR TITLE
enable-disable-integration

### PR DESCRIPTION
**Pull request by alexrupom**
- [ ] enable/disable engines

### CODE REVIEW

File: app/controllers/dradis/plugins/engines_controller.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
+module Dradis::Plugins
```
Comment (aapomm) [inline app/controllers/dradis/plugins/engines_controller.rb:1]: I don't think enabling/disabling is a dradis-plugins thing rather it should be part of dradis-pro. dradis-plugins has never had a standalone controller and adding one for it now doesn't have much merit.

File: lib/dradis/plugins/settings.rb [hunk 2]
```diff
@@ -44,6 +44,15 @@ def is_default?(key, value)
       value.to_s == @default_options[key.to_sym].to_s
     end
 
+    def engine_enable
```
Comment (aapomm) [inline lib/dradis/plugins/settings.rb:47]: https://github.com/rubocop/ruby-style-guide#bool-methods-qmark
```suggestion
    def enabled?
```

File: lib/dradis/plugins/settings.rb [hunk 3]
```diff
@@ -44,6 +44,15 @@ def is_default?(key, value)
...
+      enable = db_setting(:enable)
+      enable ? (enable == '1') : nil
```
Comment (aapomm) [inline lib/dradis/plugins/settings.rb:49]: There's some weirdness around checking for the `1` string and it doesn't help with the readability of the code. What if we just save the strings: "enable" and "disable" to represent the enabled and disabled engines respectively?

File: lib/dradis/plugins/settings.rb [hunk 4]
```diff
@@ -44,6 +44,15 @@ def is_default?(key, value)
...
+    end
+
+    def toggle_engine(value)
```
Comment (aapomm) [inline lib/dradis/plugins/settings.rb:52]: It seems better to implement an `enable` and `disable` method instead to make it more consistent with the app. See https://github.com/secroots/dradis-pro/blob/develop/app/models/user.rb#L133

File: spec/settings_spec.rb [hunk 5]
```diff
@@ -41,7 +42,6 @@ class TestEngine < ::Rails::Engine
   it "saves to db and returns persisted values" do
     expect(TestEngine::settings.host).to eq('localhost')
     TestEngine::settings.host = '127.0.0.1'
-    expect_any_instance_of(TestEngine::settings::send(:configuration_class)).to receive(:update_attribute)
```
Comment (alexrupom) [inline spec/settings_spec.rb:44]: configuration_class method was removed [here](https://github.com/dradis/dradis-plugins/pull/2/files#diff-5773e9b94584306186f2e4ad96db8100d666a97f15b78ca3c464b928760cafca) seems like this test is not needed anymore.

File: spec/engine_spec.rb [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    end
+  end
+  describe '#enable!' do
+    it 'sets enabled to true' do
+      expect { TestEngine.enable! }.to change {
+        TestEngine.enabled?
+      }.from(false).to(true)
+    end
+  end
+  describe '#disalbe!' do
```
Comment (aapomm) [inline spec/engine_spec.rb:26]:
```suggestion
  describe '#disable!' do
```

File: lib/dradis/plugins.rb [hunk 7]
```diff
@@ -11,10 +11,15 @@ def list
...
+      def enabled_list
+        @@extensions.select(&:enabled?)
+      end
+
       # Filters the list of plugins and only returns those that provide the
-      # requested feature.
+      # requested feature and enabled
       def with_feature(feature)
-        @@extensions.select do |plugin|
+        @@extensions.select(&:enabled?).select do |plugin|
```
Comment (aapomm) [inline lib/dradis/plugins.rb:22]: Why not use `enabled_list` here?

Comment (alexrupom) [reply lib/dradis/plugins.rb]: done

File: spec/settings_spec.rb [hunk 8]
```diff
@@ -21,6 +21,7 @@ class TestEngine < ::Rails::Engine
   it "sets and return default values" do
     expect(TestEngine::settings.host).to eq('localhost')
     expect(TestEngine::settings.port).to eq(80)
+    expect(TestEngine::settings.enabled).to eq(false)
```
Comment (etdsoft) [inline spec/settings_spec.rb:24]: :alpha!

**Pull request merged**